File: Src/Syntax/Syntax/NamespaceDecl.cs
using Citron.Collections;
using Pretune;

namespace Citron.Syntax
{
    [AutoConstructor, ImplementIEquatable]
    public partial class NamespaceDecl : ISyntaxNode
    {
        public ImmutableArray<string> Names { get; } // dot seperated names, NS1.NS2
        public ImmutableArray<NamespaceElement> Elements { get; }
    }
    
}

File: src/Syntax/NamespaceDecl.cpp
#include "pch.h"
#include "Syntax/NamespaceDecl.h"
#include "Syntax/NamespaceElements.h"

using namespace std;

namespace Citron::Syntax {

NamespaceDecl::NamespaceDecl(vector<string> names, vector<NamespaceElement> elements)
    : names(std::move(names)), elements(std::move(elements))
{
}

IMPLEMENT_DEFAULTS_DEFAULT(NamespaceDecl)

}